lib/puppet/network/http/factory.rb

```diff
@@ -37,6 +39,7 @@ def create_connection(site)
     http.use_ssl = site.use_ssl?
     http.read_timeout = Puppet[:http_read_timeout]
     http.open_timeout = Puppet[:http_connect_timeout]
+    http.keep_alive_timeout = KEEP_ALIVE_TIMEOUT if http.respond_to?(:keep_alive_timeout)
```

Comment:
Minor nit, but shouldn't this be looking for `http.respond_to?(:keep_alive_timeout=)` since that's the method it's going to be calling?

Reply:
Updated